core/iwasm/compilation/aot_emit_function.c

```diff
@@ -857,6 +857,14 @@ aot_compile_op_call(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
                 goto fail;
             }
 
+#if WASM_ENABLE_THREAD_MGR != 0
+            /* Insert suspend check point */
+            if (comp_ctx->enable_thread_mgr) {
+                if (!check_suspend_flags(comp_ctx, func_ctx))
+                    return false;
+            }
+#endif
```

Comment:
If to make the similar changes like interpreter for AOT, the place to add the check should be:
the end of aot_compile_op_call (after returning from the jit function):
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_function.c#L1001
and the end of aot_compile_op_call_indirect (after returning from the jit function):
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_function.c#L1647

Note that the suspend flags was also checked at the beginning of the two functions, or, before calling the jit function:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_function.c#L641
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_function.c#L1447

But adding the new checks may impact the performance of AOT/JIT when multi-thread is enabled, can we make sure that we exactly need to add these checks (both for interpreter and JIT/AOT)?

Reply:
> Note that the suspend flags was also checked at the beginning of the two functions, or, before calling the jit function:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_function.c#L641
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_function.c#L1447

> But adding the new checks may impact the performance of AOT/JIT when multi-thread is enabled, can we make sure that we exactly need to add these checks (both for interpreter and JIT/AOT)?

I tested it and it doesn't work without adding the checks at L1001 and L1647. We need to check for termination right after the function call, otherwise the next instruction will be fetched and we want to avoid that.

Reply:
OK, let's add them.